doc/api/net.md

```diff
@@ -533,8 +713,13 @@ connection is destroyed no further data can be transferred using it.
 added: v0.1.90
 -->
 
-Half-closes the socket. i.e., it sends a FIN packet. It is possible the
-server will still send some data.
+* `data` {string|Buffer}
+* `encoding` {string} A character encoding
+
+The `socket.end()` method causes a `FIN` packet to be sent and the [Writable][]
+side of the socket to be closed. This puts to socket into a "half-closed"
+state. In this state, it is possible for data to be received via the
+[Readable][] side of the socket.
```

Comment:
Note that node servers don't support sockets being half-closed by default, see `allowHalfOpen`. <---- gotcha